### PR TITLE
Update Model Component in DG

### LINKED ISSUE #245: Update Model Component in DG

### PR DESCRIPTION
**Pull request by Chandan8186**
Fix #245 

This PR updates the Model component as well as its relevant sections in the DG. It also fixes minor bugs in the UI/Logic sections.

The proposed undo/redo and archiving features from AB3 have been removed.

### CODE REVIEW

File: docs/images/ProjectClassDiagram.png
Comment (antonTan96) [inline docs/images/ProjectClassDiagram.png:1]: `ProjectPriority` should be changed to `Priority`

**Review by antonTan96 — CHANGES_REQUESTED** [review]
Other than the minor change in the project class diagram, LGTM!

File: docs/images/ProjectClassDiagram.png
Comment (Chandan8186) [reply docs/images/ProjectClassDiagram.png:1]: Thanks, I've fixed it

**Review by antonTan96 — APPROVED** [review]
LGTM!

**Pull request merged**